Modules/syslogmodule.c

```diff
@@ -193,20 +220,11 @@ syslog_syslog_impl(PyObject *module, int group_left_1, int priority,
         return NULL;
     }
 
-    /*  if log is not opened, open it now  */
-    if (!S_log_open) {
-        PyObject *openlog_ret = syslog_openlog_impl(module, NULL, 0, LOG_USER);
-        if (openlog_ret == NULL) {
-            return NULL;
-        }
-        Py_DECREF(openlog_ret);
+    _syslog_state *state = get_syslog_state(module);
+    if (state->log_open != 1 && is_main_interpreter() < 0) {
+        PyErr_SetString(PyExc_RuntimeError, "unable to use syslog.syslog at non-main interpreter.");
+        return NULL;
     }
```

Comment:
We still want the implicit open to work in the main interpreter.

Reply:
Also, `state->log_open` will always be `0` in a subinterpreter.  Instead, we have to know what the value is in the main interpreter.  For that we should probably keep the existing global variable around (and add a comment up there next to it indicating that it is only set in the main interpreter).

```suggestion
    if (!S_log_open) {
        if (is_main_interpreter() < 0) {
           PyErr_SetString(PyExc_RuntimeError, "unable to use syslog.syslog at non-main interpreter.");
           return NULL;
       }
       ...
   }
```

One might be tempted to add a global lock to prevent races between the main interpreter setting `S_log_open` and a subinterpreter reading the variable.  However, the race is unlikely and the consequences are extremely benign, while setting up a global lock is non-trivial (currently).  It might be a different story if we had a resolution on the [related DPO thread](https://discuss.python.org/t/20668/25) but we don't (yet).